addons/sale_crm/models/crm_lead.py

```diff
@@ -101,3 +101,14 @@ def _merge_get_fields_specific(self):
         # add all the orders from all lead to merge
         fields_info['order_ids'] = lambda fname, leads: [(4, order.id) for order in leads.order_ids]
         return fields_info
+
+    def _update_revenues_from_so(self):
+        for opportunity in self:
+            current_order_id = self.env['sale.order'].browse(self.env.context.get('order_id'))
+            if (
+                opportunity.expected_revenue < current_order_id.amount_untaxed
```

Comment:
In case people have False/null ? not sure if automatically converted to 0
```suggestion
                (opportunity.expected_revenue or 0) < current_order_id.amount_untaxed
```